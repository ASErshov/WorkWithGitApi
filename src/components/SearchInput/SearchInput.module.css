.submit{
    background-color: #1976D2 !important;
    color: #FFFFFF !important;
    box-shadow: none !important;
}
.submit:hover{
    background-color: #1976D2 !important;
}
.form{
    display: flex;
    flex-wrap: nowrap;
}