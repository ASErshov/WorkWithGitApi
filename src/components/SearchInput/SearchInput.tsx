import * as React from 'react'
import styles from './SearchInput.module.css'
import cn from 'clsx'
import { FormControl, OutlinedInput, InputAdornment, InputLabel, Button } from '@material-ui/core'

export type SearchInputProps = {
    handleSearch(text:string): void,
    lable?: string
    lableWidth?: number
    buttonText: string
    className?:string
}

const SearchInput: React.FC<SearchInputProps> = ({handleSearch, lable, lableWidth = 60, buttonText, className}): React.ReactElement =>{
    const [text, setText] = React.useState('');
    
    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
        setText(event.target.value as string);
    }

    const handleSubmit = (text: string) =>{
        return ()=>{handleSearch(text)}
    }

    return(
        <div className={cn(styles.form, className)}>
        <FormControl fullWidth variant="outlined" size='small'>
        {lable && <InputLabel htmlFor="text">{lable}</InputLabel>}
        <OutlinedInput
            id="text"
            startAdornment={<InputAdornment position="start"></InputAdornment>}
            labelWidth={lableWidth}
            value = {text}
            onChange = {handleChange}
        />
        </FormControl>
        <Button variant="contained" type='submit' className={styles.submit} onClick={handleSubmit(text)}>
            {buttonText}
        </Button>
        </div>
    )
}

export default SearchInput