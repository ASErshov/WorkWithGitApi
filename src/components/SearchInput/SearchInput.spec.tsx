import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import propStyles from '../../features/MainPage/MainPage.module.css'

import SearchInput from './index'

describe('<SearchInput />', () => {
  const onClick = jest.fn()
  const Component = (
    <SearchInput 
      handleSearch={onClick} 
      lable={'Поиск по слову'}
      lableWidth={145} 
      buttonText={'Поиск'} 
      className={propStyles.search}  
    />
  )

  it('Snapshot is correct', () => {
    const wrapper = renderer.create(Component).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('Passed props is valid', () => {
    const wrapper = mount(Component)

    expect(wrapper.props().lable).toEqual('Поиск по слову')
    expect(wrapper.props().lableWidth).toEqual(145)
    expect(wrapper.props().buttonText).toEqual('Поиск')
  })

  it('Passed events is valid', () => {
    const wrapper = mount(React.cloneElement(Component))

    wrapper.find('form').simulate('submit')

    expect(onClick.mock.calls.length).toBe(1)

    wrapper.find('form').simulate('submit')
    wrapper.find('form').simulate('submit')

    expect(onClick.mock.calls.length).toBe(3)
  })

  it('JSX has been rendered is valid', () => {
    const wrapper = mount(Component)
    expect(wrapper.find('.MuiButton-label').text()).toEqual('Поиск')
    expect(wrapper.find('form').hasClass(propStyles.search)).toBeTruthy()
  })
})
