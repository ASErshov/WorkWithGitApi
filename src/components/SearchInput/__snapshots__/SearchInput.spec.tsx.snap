// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> Snapshot is correct 1`] = `
<form
  action="/#"
  autoComplete="off"
  className="form search"
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink={true}
      htmlFor="text"
    >
      Поиск по слову
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
      />
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        id="text"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-2"
          style={
            Object {
              "width": 116.75,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained submit"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label"
    >
      Поиск
    </span>
  </button>
</form>
`;
